### PR TITLE
Remove period from end of `category_filter_placeholder`

### PR DESCRIPTION
**Pull request by hughlilly**
I don't believe there should be a period at the end of the placeholder text in the filter box (currenly "Search for a topic." in English).

This PR removes the period, in three commits:

1. In `en` and `en-gb` scripts
2. In most other left-to-right/Western scripts (including Cyrillic ones)
3. In all other (right-to-left) scripts

Thanks
Hugh

### CODE REVIEW

File: chstrings/qqq.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
 	"back": "{{Identical|Back}}",
 	"button_next": "Label on the button that browses to the next article.\n{{Identical|Next}}",
 	"button_wikilink": "Label on the button that links to the article so the user can add a reference.",
-	"category_filter_placeholder": "The placeholder text in the input box for filtering articles by category.",
+	"category_filter_placeholder": "The placeholder text in the input box for filtering articles by category",
```
Comment (eggpi) [inline chstrings/qqq.json:14]: I guess this one can stay, right? It's not shown anywhere, it's just a description of what the string is for translators.

**Review by eggpi — CHANGES_REQUESTED** [review]
Thank you! I've just left one small comment.

I don't have a problem merging this, but in order to make an informed decision, I'm wondering what's the reasoning behind the change. Is this some English grammar rule I'm not aware of? Or some accessibility reason?

Also note that this is not the only placeholder with a period (there's at least [`select_articles_placeholder`](https://github.com/eggpi/citationhunt/blob/master/chstrings/en.json#L46). I imagine we also need to change those for consistency?

File: chstrings/qqq.json (hunk 1 above)
Comment (hughlilly) [reply chstrings/qqq.json:14]: Oh yes, sorry, that got scooped up by my overzealous regex search; I'll restore the period.

**Comment by hughlilly** [comment]
> I don't have a problem merging this, but in order to make an informed decision, I'm wondering what's the reasoning behind the change. Is this some English grammar rule I'm not aware of? Or some accessibility reason?

I think it's more that placeholder text, at least as I've seen it most places, isn't typically a full sentence; it's intended as a prompt to the user, as opposed to an explanation of some part of the interface. Does that make sense?

(As for accessibility, [some warn against the use of the `placeholder` attribute entirely](https://www.nngroup.com/articles/form-design-placeholders/). In the case at hand the placeholder doesn't disappear when the field is clicked on, so at least one of the downsides mentioned in that article is not relevant.)

>Also note that this is not the only placeholder with a period (there's at least select_articles_placeholder). I imagine we also need to change those for consistency?

Yes, this text would need to change in a similar way, in my opinion.

File: chstrings/qqq.json (hunk 1 above)
Comment (hughlilly) [reply chstrings/qqq.json:14]: Actually, if this is used in a form-field `placeholder` value, it should not end in a period. Where does this text show up?

Comment (eggpi) [reply chstrings/qqq.json:14]: It's only visible in TranslateWiki, the tool that volunteers use to translate these strings. It is an explanation (to the translator) of where the string will eventually be used, so it's not used in an actual form. See [here](https://translatewiki.net/w/i.php?title=Special:Translate&showMessage=citationhunt-category_filter_placeholder&group=citationhunt&language=fr&filter=&optional=1&action=translate) for a concrete example.

**Comment by eggpi** [comment]
> I think it's more that placeholder text, at least as I've seen it most places, isn't typically a full sentence; it's intended as a prompt to the user, as opposed to an explanation of some part of the interface. Does that make sense?

Sure, sounds good. I also noticed that English Wikipedia uses "Search Wikipedia" (no period) in its search bar, for what it's worth.
 
> (As for accessibility, [some warn against the use of the `placeholder` attribute entirely](https://www.nngroup.com/articles/form-design-placeholders/). In the case at hand the placeholder doesn't disappear when the field is clicked on, so at least one of the downsides mentioned in that article is not relevant.)

Good read, thanks for the link!

> > Also note that this is not the only placeholder with a period (there's at least select_articles_placeholder). I imagine we also need to change those for consistency?
> 
> Yes, this text would need to change in a similar way, in my opinion.

Would you like to help with this too, please? If so, let me know if you want to raise a separate PR or just add to this one please. I'm fine with the change either way :)

**Comment by hughlilly** [comment]
> > > Also note that this is not the only placeholder with a period (there's at least select_articles_placeholder). I imagine we also need to change those for consistency?
> > 
> > 
> > Yes, this text would need to change in a similar way, in my opinion.
> 
> Would you like to help with this too, please? If so, let me know if you want to raise a separate PR or just add to this one please. I'm fine with the change either way :)

I've begun to update this variable in the same way; could you take a look at the commit and let me know if you want me to continue with the other language variants?

Thanks
Hugh

**Comment by hughlilly** [comment]
> Sure, sounds good. I also noticed that English Wikipedia uses "Search Wikipedia" (no period) in its search bar, for what it's worth.

Yes; it's also sensible to use an ellipsis, as [this demo for `textarea` does](https://www.w3schools.com/tags/att_placeholder.asp). Again that idea that it's an instruction or prompt to the user to take some action, not a statement in an of itself. The [MDN Docs example](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#attr-placeholder) similarly does not end in a period.

**Comment by eggpi** [comment]
Yep, 7a4350b6275910471b1ef1ea13fcb3bbe89d4b21 looks good, please go ahead.

FYI, the form for this placeholder it's part of the ["Customize" feature ](https://meta.wikimedia.org/wiki/Citation_Hunt#Customize) and appears when the user chooses to search for articles within Citation Hunt. Just so you know where the change will appear :)

**Comment by hughlilly** [comment]
> FYI, the form for this placeholder it's part of the ["Customize" feature ](https://meta.wikimedia.org/wiki/Citation_Hunt#Customize) and appears when the user chooses to search for articles within Citation Hunt. Just so you know where the change will appear :)

Ah yes, I managed to find that; thanks. 😎

**Comment by eggpi** [comment]
Thank you!

**Pull request merged**